File: scripts/check_database_schema.mjs
#!/usr/bin/env node

import { createClient } from "@supabase/supabase-js";

// Use the environment variables from .env
const supabaseUrl = "https://vaypgzvivahnfegnlinn.supabase.co";
const supabaseKey =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZheXBnenZpdmFobmZlZ25saW5uIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTc4NzEzMjIsImV4cCI6MjA3MzQ0NzMyMn0.RISVGvci0v8GD1DtmnOD9lgJSYyfErDg1c__24K82ws";

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkDatabaseSchema() {
  console.log("🔍 Checking current database schema...\n");

  try {
    // Test a simple query first to see what columns exist
    console.log("📊 Testing current columns...");
    const { data, error } = await supabase.from("games").select("*").limit(1);

    if (error) {
      console.error("❌ Error querying games table:", error.message);
      return;
    }

    if (data && data.length > 0) {
      console.log("✅ Games table accessible");
      console.log("📋 Available columns:", Object.keys(data[0]));

      // Check if game_id column exists
      if ("game_id" in data[0]) {
        console.log("✅ game_id column exists");
      } else {
        console.log("❌ game_id column is missing");
        console.log("💡 Need to add game_id column to match API expectations");
      }
    } else {
      console.log("⚠️ Games table is empty");
    }

    // Test the problematic query that was failing
    console.log("\n🧪 Testing API query with game_id...");
    const { data: apiData, error: apiError } = await supabase
      .from("games")
      .select(
        `
        id,
        home_team,
        away_team,
        game_time,
        status,
        week,
        season,
        home_score,
        away_score,
        quarter,
        time_remaining
      `
      )
      .order("game_time", { ascending: true })
      .limit(1);

    if (apiError) {
      console.error("❌ API query failed:", apiError.message);
    } else {
      console.log("✅ API query successful with updated column names");
      if (apiData && apiData.length > 0) {
        console.log("📄 Sample game data:", apiData[0]);
      }
    }
  } catch (error) {
    console.error("❌ Database check failed:", error.message);
  }
}

checkDatabaseSchema();

File: scripts/check_database_schema.py
#!/usr/bin/env python3
"""Check what tables and columns actually exist in Supabase"""

import os
from dotenv import load_dotenv
from supabase import create_client

load_dotenv()

SUPABASE_URL = os.getenv('VITE_SUPABASE_URL')
SUPABASE_KEY = os.getenv('VITE_SUPABASE_ANON_KEY')

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

print("🔍 Checking Supabase Schema\n")

# Try to query expert_episodic_memories to see what columns exist
try:
    result = supabase.table('expert_episodic_memories').select('*').limit(1).execute()
    print("✅ Table 'expert_episodic_memories' exists")

    if result.data and len(result.data) > 0:
        print(f"\nColumns in table:")
        for key in result.data[0].keys():
            print(f"  • {key}")
    else:
        print("\n📋 Table is empty - trying to get schema from error...")

        # Try to insert with a field we know exists
        try:
            test_insert = supabase.table('expert_episodic_memories').insert({
                'test': 'value'
            }).execute()
        except Exception as e:
            error_str = str(e)
            print(f"\n❌ Insert error (expected): {error_str[:200]}")

except Exception as e:
    error_str = str(e)
    if 'does not exist' in error_str.lower():
        print("❌ Table 'expert_episodic_memories' does NOT exist")
        print("\n⚠️  Migration 011_expert_episodic_memory_system.sql needs to be applied!")
    else:
        print(f"❌ Error: {error_str}")

# Check other tables
print("\n\n🔍 Checking other tables:")

for table in ['expert_reasoning_chains', 'expert_belief_revisions', 'expert_learned_principles']:
    try:
        result = supabase.table(table).select('id').limit(1).execute()
        print(f"✅ {table}: exists")
    except Exception as e:
        if 'does not exist' in str(e).lower():
            print(f"❌ {table}: DOES NOT EXIST")
        else:
            print(f"⚠️  {table}: {str(e)[:100]}")

print("\n\n💡 RECOMMENDATION:")
print("Run migration: supabase/migrations/011_expert_episodic_memory_system.sql")
print("Or apply via Supabase Dashboard: SQL Editor → New Query → Paste migration → Run")